token/rust/src/token.rs

```diff
@@ -97,40 +97,39 @@ where
     }
 
     /// Create and initialize a token.
+    #[allow(clippy::too_many_arguments)]
     pub async fn create_mint<'a, S2: Signer>(
         client: Arc<dyn ProgramClient<T>>,
         payer: S,
         mint_account: &'a S2,
         mint_authority: &'a Pubkey,
         freeze_authority: Option<&'a Pubkey>,
         decimals: u8,
+        extension_types: &'a [ExtensionType],
+        extension_instructions: &'a [Instruction],
```

Comment:
Longer term, I don't love the ergonomics of the user/client having to create the extension instructions themselves; ix/tx building is one of the main purposes of the rust client, imo.
Perhaps we move to having a client specific enum that holds ExtensionTypes+required init data?
That can be a separate PR, however.

Reply:
That makes a lot of sense to me.  I'm down to have a separate enum for extension init data!  If we make separate types for each set of init data, and then bundle them all into a separate enum, it should be pretty slick.